app/scripts/lib/nonce-tracker.js

```diff
@@ -83,6 +92,14 @@ class NonceTracker {
     return mutex
   }
 
+  _getLocalNonce (address) {
+    const confirmedTransactions = this.getConfirmedTransactions(address)
```

Comment:
I was thinking we'd want the highest nonce of all non-failed transactions (so including submitted txs)